File: jpa-fetch-join/src/main/java/com/example/jpafetchjoin/order/OrderApi.java
package com.example.jpafetchjoin.order;

import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/orders")
@RequiredArgsConstructor
public class OrderApi {

  private final OrderRepository orderRepository;

  @GetMapping
  public Page<Order> getOrderAll(Pageable pageable){
    return orderRepository.findAll(pageable);
  }

  @GetMapping("/all")
  public Page<Order> getOrders(Pageable pageable){
    return orderRepository.findByPageWithAll(pageable);
  }

  @GetMapping("/address")
  public Page<Order> getOrdersWithAddress(Pageable pageable){
    return orderRepository.findByPageWithAddress(pageable);
  }

  @GetMapping("/products")
  public Page<Order> getOrdersWithProducts(Pageable pageable){
    return orderRepository.findByPageWithProducts(pageable);
  }

}

File: application-event/src/main/kotlin/com/example/applicationevent/order/OrderApi.kt
package com.example.applicationevent.order

import com.example.applicationevent.item.Item
import org.springframework.web.bind.annotation.*


@RestController
@RequestMapping("/orders")
class OrderApi(
        private val orderRepository: OrderRepository,
        private val orderService: OrderService
) {

    @GetMapping
    fun getOrders(): MutableList<Order> {
        return orderRepository.findAll()
    }

    @PostMapping
    fun ordering(@RequestBody itemCode: List<String>): Order {
        return orderService.order(itemCode)
    }


}